### ISSUE TITLE
JPEG image gets PNG extension due to caching issue

### ISSUE THREAD

**Issue by nibe** [gohugoio/hugo#11856]
There seems to be a caching issue resulting in a wrong filename extension.

```html
{{ $original := .Resources.Get "test.jpg" }}
{{- $png := $original.Resize "400x png" }}
{{- $jpg := $png.Resize "200x jpg" }}
<img src="{{ $jpg.RelPermalink }}" width="{{ $jpg.Width }}" height="{{ $jpg.Height }}">
```

First I get:
`<img src="/test_hu5be2162292b2ff441686d7cf4a0b90a6_2368_ac11a309a903d236d4e1c9c7856834da.jpg" width="200" height="200">`

All good. But after restarting `hugo server` or rerunning `hugo`:
`<img src="/test_hu5be2162292b2ff441686d7cf4a0b90a6_2368_c1af3cf7e437c7562ad60da90751875e.png" width="200" height="200">`

Why `.png`? The actual format seems to be JPEG but the filename extension is wrong and the filename different. It seems to happen with any JPEG image. Conversion to `webp` seems to work fine. To get `.jpg` again in the next build:
`rm -rf resources/_gen`

### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.121.1-00b46fed8e47f7bb0a85d7cfc2d9f1356379b740+extended darwin/arm64 BuildDate=2023-12-08T08:47:45Z VendorInfo=brew
</pre>

### Does this issue reproduce with the latest release?

Yes.

**Comment by XoL1507** [comment]
crates/sui-graphql-rpc/src/server/graphiql_server.rs

**Comment by XoL1507** [comment]
[]()

**Comment by XoL1507** [comment]
_/_

**Comment by jmooring** [comment]
Yeah, this is broken, somewhat related to <https://github.com/gohugoio/hugo/issues/6943>.

Do this instead:

```text
{{ $original := .Resources.Get "test.jpg" }}
{{- $png := $original.Resize "400x png" }}
{{- $jpg := $original.Process "resize 200x jpg" }}
<img src="{{ $jpg.RelPermalink }}" width="{{ $jpg.Width }}" height="{{ $jpg.Height }}">
```

**Comment by jmooring** [comment]
This was fixed in v0.155.1, released 3 Jan 2026. See https://github.com/gohugoio/hugo/commit/b5d43cdc1783796d9c6b17c7e135fa46d8b0279d.
